pygama/dsp/_processors/pulse_injector.py

```diff
@@ -0,0 +1,28 @@
+import numpy as np
+from numba import guvectorize
+from math import log,exp
+
+@guvectorize(["void(float32[:], float32, float32, float32, float32, float32[:])",
+              "void(float64[:], float64, float64, float64, float64, float64[:])"],
+              "(n),(), (), (), ()->(n)", nopython=True, cache=True)
+def inject_sig_pulse(wf_in,t0,rise,amp,tau,wf_out):
+    """
+    Inject sigmoid pulse into existing waveform to simulate pileup
+    """    
+    wf_out[:] = wf_in[:]
+    for T in range(len(wf_out)):

```

Comment:
why don't you start the loop from T=t0?

Reply:
For injecting pre trigger pileups I use a negative time, so starting from t0 would start to use negative indices for adding the pulse to the existing waveform and would produce incorrect results. I Could use an if statement to check the current index is positive before adding the pulses at that point, or maybe there's another work around. Not sure what the preferred solution would be here.